docs/how-clerk-works/tokens-signatures.mdx

```diff
@@ -0,0 +1,84 @@
+---
+title: Tokens and signatures
+description: Learn about JWTs, digital signatures, and how Clerk uses them to securely authenticate users.
+---
+
+## Digital signatures
+
+Digital signatures are a cryptographic technique that ensures the authenticity and integrity of messages. They guarantee that:
+
+1. The message originates from a specific sender (authenticity).
+1. The message's content hasn't been modified from how it was written by the sender (integrity).
+
+However, digital signatures do **not** encrypt the message—anyone can read its contents.
+
+### How digital signatures work
+
+Digital signatures use **public key cryptography**, which involves a key pair: a private key (kept secret) and a public key (shared openly). Here's the process:
+
+1. A private key is used to create a **signature**:
+   1. The message is [hashed](https://www.techtarget.com/searchdatamanagement/definition/hashing) (a unique, fixed-length representation of the message).
+   1. The hash is encrypted using the sender's private key, creating the signature.
+1. The message and signature are sent to the recipient.
+1. The recipient verifies the signature using the sender's public key by:
+   1. Decrypting the signature to retrieve the hash.
+   1. Independently hashing the message and comparing the two hashes. If they match, the message is authentic and untampered.
+
+So for example, you get a publicly readable message, like `hello world`, and the message is “signed” with the signature `j2e80w8dj9f8`. If you'd like to be sure that the message is genuine, and you have a copy of the sender's public key, you can use this to decrypt the signature and make sure that it works. If it does, you know for sure who sent the message, and that nobody intercepted the message and messed with it in the middle, since part of the verification process involves hashing the message and comparing it to the decrypted signature.
```

Comment:
```suggestion
So for example, imagine you get a publicly readable message, like `hello world`, and the message is “signed” with the signature `j2e80w8dj9f8`. If you'd like to be sure that the message is genuine, and you have a copy of the sender's public key, you can use this key to decrypt the signature and make sure that it is valid. If it is, you know two things for sure: who sent the message, and that nobody intercepted the message and messed with it in the middle, since part of the verification process involves hashing the message and comparing it to the decrypted signature.
```